core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/TokenUtils.scala

```diff
@@ -0,0 +1,132 @@
+  }
+
+  def checkTokenValid(token: String): Boolean = {
+    if (token == null || token.isEmpty) {
+      false
+    }
+    try{
+      val tokenParser =   new FabricTokenParser(token)
+      val expiryEpoch = tokenParser.getExpiry
+      val now = Instant.now().getEpochSecond
+      now < expiryEpoch - 60
+    } catch {
+      case e: InvalidJwtTokenException =>
+        SynapseMLLogging.logMessage(s"TokenUtils::checkTokenValid: Token used to trigger telemetry " +
+          s"endpoint is invalid. Exception = $e")
+        false
+      case e: JwtTokenExpiryMissingException =>
+        SynapseMLLogging.logMessage(s"TokenUtils::checkTokenValid: Token misses expiry. " +
+          s"Exception = $e")
+        false
+    }
+  }
+
+  private def refreshAccessToken(): Unit = {
+    try {
+      if (SparkContext.getOrCreate() != null) {
+        val token = getAccessToken("pbi")
+        AADToken = token
+      } else {
+        val token = new FabricTokenServiceClient().getAccessToken("pbi")
+        AADToken = token
+      }
+    } catch {
+      case e: Exception =>
+        SynapseMLLogging.logMessage(s"refreshAccessTok: failed to refresh pbi tok. Exception: {$e}. (usage test)")
+    }
+  }
+
+  def getMWCToken(shared_host: String, WorkspaceId: String, capacity_id: String,
+                    workload_type: String): MwcToken = {
+    val url: String = shared_host + "/metadata/v201606/generatemwctokenv2"
+
+    val payLoad = s"""{
+      |"capacityObjectId": "$capacity_id",
+      |"workspaceObjectId": "$WorkspaceId",
+      |"workloadType": "$workload_type"
+    }""".stripMargin
+
+    val driverAADToken = getAccessToken
+
+    val headers = Map(
+      "Content-Type" -> "application/json",
+      "Authorization" -> s"""Bearer $driverAADToken""".stripMargin,
+      "x-ms-workload-resource-moniker" -> UUID.randomUUID().toString
+    )
+
+    try{
+      val response = usagePost(url, payLoad, headers)
+      var targetUriHost = response.asJsObject.fields("TargetUriHost").convertTo[String]
+      targetUriHost = s"https://$targetUriHost"
+      response.asJsObject.fields.updated("TargetUriHost", targetUriHost)
+
+      implicit val mwcTokenFormat: RootJsonFormat[MwcToken] = jsonFormat3(MwcToken)
+      response.convertTo[MwcToken]
+    }
+    catch {
+      case e: NoSuchElementException =>
+        SynapseMLLogging.logMessage(s"TokenUtils.getMWCToken: Cannot retrieve targetUriHost from MWC Token.")
+        throw e
+      case e: DeserializationException =>
+        SynapseMLLogging.logMessage(s"TokenUtils.getMWCToken: The structure of response is not of type MwcToken.")
+        throw e
+      case e: ParsingException =>
+        SynapseMLLogging.logMessage(s"TokenUtils.getMWCToken: The structure of json response is formed correctly.")
+        throw e
+      case e: Exception =>
+        SynapseMLLogging.logMessage(s"getMWCTok: Failed to fetch MWC token that is required to " +
+          s"get cluster details: $e.")
+        throw e
```

Comment:
we already have generic methods to log error messages, perhaps use those instead of cluttering this code with a loop that just catches and rethrows. See the logError function and how its used in SynapseMlLogging

Reply:
@mhamilton723 Did you mean don't case within catch and just use SynapseMLLogging.logErrorBase?